pkg/runtime/helper_test.go

```diff
@@ -63,6 +63,28 @@ func TestExtractList(t *testing.T) {
 	}
 }
 
+func TestExtractGenericList(t *testing.T) {
```

Comment:
Can you add a few more test cases? Test []*runtime.Object, []api.Foo, []*api.Foo